Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -41,26 +41,26 @@ extension KafkaProducerCloseOnTerminate: NIOAsyncSequenceProducerDelegate {
     }
 }
 
-// MARK: - KafkaMessageAcknowledgements
+// MARK: - KafkaProducerEvents
 
-/// `AsyncSequence` implementation for handling messages acknowledged by the Kafka cluster (``KafkaAcknowledgedMessage``).
-public struct KafkaMessageAcknowledgements: AsyncSequence {
-    public typealias Element = Result<KafkaAcknowledgedMessage, KafkaAcknowledgedMessageError>
+/// `AsyncSequence` implementation for handling ``KafkaProducerEvent``s emitted by Kafka.
+public struct KafkaProducerEvents: AsyncSequence {
+    public typealias Element = KafkaProducerEvent
     typealias BackPressureStrategy = NIOAsyncSequenceProducerBackPressureStrategies.NoBackPressure
     typealias WrappedSequence = NIOAsyncSequenceProducer<Element, BackPressureStrategy, KafkaProducerCloseOnTerminate>
     let wrappedSequence: WrappedSequence
 
-    /// `AsynceIteratorProtocol` implementation for handling messages acknowledged by the Kafka cluster (``KafkaAcknowledgedMessage``).
-    public struct AcknowledgedMessagesAsyncIterator: AsyncIteratorProtocol {
+    /// `AsynceIteratorProtocol` implementation for handling ``KafkaProducerEvent``s emitted by Kafka.
+    public struct KafkaProducerEventsAsyncIterator: AsyncIteratorProtocol {
```

Comment:
I'm not sure what conventions we're using for naming here, so you can ignore this comment if it goes against what we've been doing elsewhere in this project, but since this struct is nested inside `KafkaProducerEvents` already, I would skip that prefix and name this `AsyncIterator`.
`KafkaProducerEvents.AsyncIterator` seems cleaner than `KafkaProducerEvents.KafkaProducerEventsAsyncIterator`.